kclvm/tools/src/LSP/src/state.rs

```diff
@@ -208,28 +199,48 @@ impl LanguageServerState {
             match get_file_name(vfs, file.file_id) {
                 Ok(filename) => {
-                    match parse_param_and_compile(
-                        Param {
-                            file: filename.clone(),
-                            module_cache: self.module_cache.clone(),
-                        },
-                        Some(self.vfs.clone()),
-                    ) {
-                        Ok((prog, scope, diags, gs)) => {
-                            let end = start.elapsed();
-                            self.log_message(format!("compile time: {:?}s", end.as_secs_f32()));
-                            self.analysis.set_db(
-                                file.file_id,
-                                AnalysisDatabase {
-                                    prog,
-                                    scope,
-                                    diags,
-                                    gs,
-                                },
-                            );
+                    let mut snapshot = self.snapshot();
+
+                    self.thread_pool.execute({
+                        let mut snapshot = self.snapshot();
+                        let sender = self.task_sender.clone();
+                        let module_cache = self.module_cache.clone();
+                        let uri = url(&snapshot, file.file_id).unwrap();
+                        move || match parse_param_and_compile(
+                            Param {
+                                file: filename.clone(),
+                                module_cache,
+                            },
+                            Some(snapshot.vfs),
+                        ) {
+                            Ok((prog, _, diags, gs)) => {
+                                let mut db = snapshot.db.write();
+                                db.insert(
+                                    file.file_id,
+                                    AnalysisDatabase {
+                                        prog,
+                                        diags: diags.clone(),
+                                        gs,
+                                    },
+                                );
+
+                                let diagnostics = diags
+                                    .iter()
+                                    .flat_map(|diag| kcl_diag_to_lsp_diags(diag, filename.as_str()))
+                                    .collect::<Vec<Diagnostic>>();
+                                sender.send(Task::Notify(lsp_server::Notification {
+                                    method: PublishDiagnostics::METHOD.to_owned(),
+                                    params: to_json(PublishDiagnosticsParams {
+                                        uri,
+                                        diagnostics,
+                                        version: None,
+                                    })
+                                    .unwrap(),
```

Comment:
prevent the `unwrap()` code.

Reply:
It is clearly safe here， this unwrap() as same as https://github.com/kcl-lang/kcl/issues/760 and https://github.com/kcl-lang/kcl/pull/778